From 35332bd0de6bec9b1aa364643c9ca0c8bf6b6e49 Mon Sep 17 00:00:00 2001
From: Dave Young <dyoung@redhat.com>
Date: Thu, 22 May 2014 02:42:56 -0400
Subject: [firmware] efi: Simplify EFI_DEBUG

Message-id: <1400726607-13717-5-git-send-email-dyoung@redhat.com>
Patchwork-id: 80105
O-Subject: [RHEL7.1 PATCH 04/35] x86/efi: Simplify EFI_DEBUG
Bugzilla: 1080109
RH-Acked-by: Chao WANG <chaowang@redhat.com>
RH-Acked-by: Vivek Goyal <vgoyal@redhat.com>

Resolves: bz1080109

Brew build:
https://brewweb.devel.redhat.com/taskinfo?taskID=7480539

Test: Tested kexec/kdump on HP workstation and qemu.

Upstream commit:
commit f4fccac05f7f6bacb8e481a84d175e85ffcf9fe2
Author: Borislav Petkov <bp@suse.de>
Date:   Thu Oct 31 17:24:59 2013 +0100

    x86/efi: Simplify EFI_DEBUG

    ... and lose one #ifdef .. #endif sandwich.

    Signed-off-by: Borislav Petkov <bp@suse.de>
    Signed-off-by: Matt Fleming <matt.fleming@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/platform/efi/efi.c b/arch/x86/platform/efi/efi.c
index 92c0234..a9b33c6 100644
--- a/arch/x86/platform/efi/efi.c
+++ b/arch/x86/platform/efi/efi.c
@@ -51,7 +51,7 @@
 #include <asm/x86_init.h>
 #include <asm/rtc.h>
 
-#define EFI_DEBUG 1
+#define EFI_DEBUG
 
 #define EFI_MIN_RESERVE 5120
 
@@ -398,9 +398,9 @@ int __init efi_memblock_x86_reserve_range(void)
  return 0;
 }
 
-#if EFI_DEBUG
 static void __init print_efi_memmap(void)
 {
+#ifdef EFI_DEBUG
  efi_memory_desc_t *md;
  void *p;
  int i;
@@ -415,8 +415,8 @@ static void __init print_efi_memmap(void)
    md->phys_addr + (md->num_pages << EFI_PAGE_SHIFT),
    (md->num_pages >> (20 - EFI_PAGE_SHIFT)));
  }
-}
 #endif  /*  EFI_DEBUG  */
+}
 
 void __init efi_reserve_boot_services(void)
 {
@@ -696,10 +696,7 @@ void __init efi_init(void)
   x86_platform.set_wallclock = efi_set_rtc_mmss;
  }
 #endif
-
-#if EFI_DEBUG
  print_efi_memmap();
-#endif
 }
 
 void __init efi_late_init(void)
-- 
1.7.1